### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by janvanrijn**

### CODE REVIEW

File: openml_OS/models/api/v1/Api_data.php [hunk 1]
```diff
@@ -307,12 +337,10 @@ private function data_delete($data_id) {
       //}
     }
 
-    $result = $this->Dataset->delete( $dataset->did );
-    $this->Data_feature->deleteWhere('did =' . $dataset->did);
-    $this->Data_quality->deleteWhere('data =' . $dataset->did);
+    $result = $this->Dataset->delete($dataset->did);
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_data.php]: Why not delete the data features and qualities? Is deletion guaranteed by mysql cascades?

File: openml_OS/models/api/v1/Api_data.php [hunk 2]
```diff
@@ -503,6 +531,11 @@ private function status_update($data_id, $status) {
       return;
     }
     
+    if ($status == 'active' && !$this->user_has_admin_rights) {
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_data.php]: This looks a bit weird. Should a dataset not always be deactivated first, even by the admin? 
If a user wants to remove her dataset, she also has to deactivate it first?

File: openml_OS/models/api/v1/Api_data.php [hunk 3]
```diff
@@ -547,18 +580,14 @@ private function status_update($data_id, $status) {
...
     $dataset = $this->Dataset->getById($data_id);
-    if( $dataset === false ) {
+    if ($dataset === false) {
       $this->returnError(271, $this->version);
       return;
     }
 
-    if($dataset->visibility != 'public' && $dataset->uploader != $this->user_id) {
-      $this->returnError(271, $this->version); // Add special error code for this case?
+    if ($dataset->visibility != 'public' && !($dataset->uploader == $this->user_id || $this->user_has_admin_rights)) {
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_data.php]: I don't understand this. What does visibility have to do with the possibility to remove a dataset? Is it not allowed to remove a non-public dataset?

File: openml_OS/models/api/v1/Api_evaluation.php [hunk 4]
```diff
@@ -40,27 +40,29 @@ private function evaluation_request($evaluation_engine_id, $order, $segs) {
     for ($i = 0; $i < count($segs); $i += 2) {
       $query_string[$segs[$i]] = urldecode($segs[$i+1]);
       if (in_array($segs[$i], $legal_filters) == false) {
-        $this->returnError(546, $this->version, $this->openmlGeneralErrorCode, 'Legal filter operators: ' . implode(',', $legal_filters) .'. Found illegal filter: ' . $segs[$i]);
+        $this->returnError(1011, $this->version, $this->openmlGeneralErrorCode, 'Legal filter operators: ' . implode(',', $legal_filters) .'. Found illegal filter: ' . $segs[$i]);
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_evaluation.php]: Would be nice to generate an overview of error codes per API call. It becomes very hard to track like this...

File: openml_OS/models/api/v1/Api_run.php [hunk 5]
```diff
@@ -276,13 +281,8 @@ private function run_reset($run_id) {
...
     }
-
-    $result = true;
-    $result = $result && $this->Trace->deleteWhere('`run_id` = "' . $run->rid . '" ');
-    $result = $result && $this->Evaluation->deleteWhere('`source` = "' .  $run->rid. '" ');
-    $result = $result && $this->Evaluation_fold->deleteWhere('`source` = "' . $run->rid . '" ');
-    $result = $result && $this->Evaluation_sample->deleteWhere('`source` = "' . $run->rid . '" ');
-    $result = $result && $this->Run_evaluated->deleteWhere('`run_id` = "' . $run->rid . '" ');
+    
+    $result = $this->Run_evaluated->deleteWhere('`run_id` = "' . $run->rid . '" ');
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_run.php]: Same questions? Are dependent records automatically deleted?

File: openml_OS/models/api/v1/Api_data.php (hunk 1 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_data.php]: yes

File: openml_OS/models/api/v1/Api_data.php (hunk 2 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_data.php]: this basically states: only admins can activate a dataset.

File: openml_OS/models/api/v1/Api_data.php (hunk 3 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_data.php]: this ensures that only:
- features of public datasets can be seen
- admins can see non-public datasets
- users can see non-public datasets (when they own)

that's what i intended to write down. please verify if that happens.

File: openml_OS/models/api/v1/Api_evaluation.php (hunk 4 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_evaluation.php]: agreed

File: openml_OS/models/api/v1/Api_run.php (hunk 5 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_run.php]: yes.

File: openml_OS/models/api/v1/Api_data.php (hunk 3 above)
Comment (joaquinvanschoren) [reply openml_OS/models/api/v1/Api_data.php]: Ah sorry, looked at the wrong function name in the diff. Looks good.

File: openml_OS/models/api/v1/Api_data.php (hunk 2 above)
Comment (joaquinvanschoren) [reply openml_OS/models/api/v1/Api_data.php]: ok.

**Comment by joaquinvanschoren** [comment]
Shall we merge?

**Pull request merged**